src/SmoFacade/Database.cs

```diff
@@ -111,6 +113,69 @@ public List<BackupMetadata> RecentBackups()
       return backups;
     }
 
+    public decimal MostRecentFullBackupLsn()
+    {
+      var query = "SELECT MAX(database_backup_lsn) as most_recent_full_backup_lsn " +
+                  "FROM msdb.dbo.backupset " +
+                  "WHERE database_name = @dbName " +
+                  "AND is_copy_only = 0";
```

Comment:
No - we want to find the max database_backup_lsn, which could come from a Full, Diff, or Log record

Reply:
oh I see - so you are getting the highest `databaseBackup` lsn because that will be the last full backup's `checkpoint` LSN?

If so, I don't think we should be doing that.
The `databaseBackup` lsn is only used to link back to the last full-backup's checkpoint LSN to build the chain.

Also, for full backups, their `dabaseBackup` lsn doesn't have to equal their `checkpoint` lsn

Reply:
see https://github.com/factset/AgDatabaseMove/pull/97#discussion_r857472081
(I felt the conversation was being split into two threads)